### PR TITLE
Rule selector parser

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Rule selector parser

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

N/A

### CODE REVIEW

File: parsers/rule.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/RedHatInsights/insights-operator-utils/types"
+)
+
+// ParseRuleSelector function parse the rule selector which consists of
```
Comment (epapbak) [inline parsers/rule.go:25]: parses

File: parsers/rule_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+		assert.Equal(t, err.Error(), "invalid rule ID, it must contain only rule ID and error key separated by |")
+	})
+}
+
+// TestParseRuleSelectorMoreSeparators checks the function
+// parsers.ParseRuleSelector for input with more separators.
+func TestParseRuleSelectorMoreSeparators(t *testing.T) {
+	t.Run("foo|bar|baz", func(t *testing.T) {
+		_, _, err := parsers.ParseRuleSelector("")
```
Comment (epapbak) [inline parsers/rule_test.go:62]: should be "foo|bar|baz", right?

File: parsers/rule.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		return types.Component(""), types.ErrorKey(""), err
+	}
+
+	// check component name and error key content
+	IDValidator := regexp.MustCompile(`^[a-zA-Z_0-9.]+$`)
+
+	isRuleIDValid := IDValidator.MatchString(splitedRuleID[0])
+	isErrorKeyValid := IDValidator.MatchString(splitedRuleID[1])
+
+	if !isRuleIDValid || !isErrorKeyValid {
```
Comment (epapbak) [inline parsers/rule.go:43]: Seeing the tests below, maybe it's better to separate the checks and give a different error for component and error key, so it's clearer what is wrong in the input.

Comment (tisnik) [reply parsers/rule.go:43]: probably, will look at it

File: parsers/rule.go (hunk 1 above)
Comment (tisnik) [reply parsers/rule.go]: fixed

File: parsers/rule.go (hunk 3 above)
Comment (tisnik) [reply parsers/rule.go]: updated!

File: parsers/rule_test.go (hunk 2 above)
Comment (tisnik) [reply parsers/rule_test.go]: fixed

**Review by epapbak — APPROVED** [review]

**Review by dpensi — APPROVED** [review]

**Pull request merged**